### DISCUSSION TITLE
Maintainers/Codeowners - future development of the project

### DISCUSSION THREAD

**Discussion by pat-s** [kube-hetzner/terraform-hcloud-kube-hetzner#1557] [Q&A] [answered]
@mysticaltech Are there any ideas to extend the development of this project to a group of people? While you did a lot/the most for this judging from the commit log and have reasonable response times, I wonder if extending permissions for PR reviews and other actions to a group of people would help the project moving forward?

Also, the project is owned by the `kube-hetzner` org. Currently only one member is publicly visible and the overall governance is a bit unclear. 

Would be cool to shine some light on these points moving forward, for me personally and I assume more fellow users/readers :)

**Comment by mysticaltech** [comment, marked as answer]
Hey @pat-s, appreciate the input. There's a bunch of us with commit access +10, and some with merge access too. I will investigate why the team members are not showing.

For the moment, I will continue to lead the project, and we will improve time to resolution of issues.

Also, there may be some important changes in the codebase thanks to v3 that @aleksasiriski is working on. More news on that soon, when he's ready to share.

We will probably publish another branch for that first, to gather feedback.

So things are moving and we will keep on trying to make the project better and more stable.

**Reply by pat-s** [reply]
@mysticaltech I feel the need come back to this again, as right now there are several PRs waiting for review/comments/merge for several weeks - and AFAICS you are the only one which has issued official PR reviews or merges within the last year? (Didn't check all commits in detail but I couldn't find anyone else lately).

For the continuous progress of the project it would be good to have additional trusted entities which can do so when you're not available *which are also active*. You mentioned there are 10+ people with permissions, but they all seem to be inactive.

Reviews often need several rounds and if each takes several weeks, this increases the likelihood of conflicts, lowers a PR openers motivation and also makes it harder to get back into the topic. 

I think it might be time to consolidate the list of people with merge permissions so the project can get into a more continuous development state. Right now, the reality is that you block-merge almost everything ever few weeks/months, while in between nothing is happening.

In addition to that a suggestion: please switch to "squash" merges for PRs. Doing a "merge commit" and including all individual commits of PRs makes the commit history quite chaotic and also makes reverting a PR a lot harder. 

Again, please takes this as a positive recommendation which aims to benefit the project itself. Everyone is happy you're leading this and (hopefully) continue doing so! However, such a project "must" consist of multiple (trusted) people to really "work", otherwise there is a high risk for a "single-point of failure/stall", which in the end all users would suffer from :)

**Comment by aleksasiriski** [comment]
Hey there, I'm pushing the move to a better structure of the project with many modules. I've stopped helping with issues and PRs on the current version of the project because of that since much of my time has been taken by the rewrite and work, but after the release of that I will continue with supporting the project.
